src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const resultArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties' :
+        Object.assign(stateCopy, action.extraData);
+        addNewState(stateCopy);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        addNewState(stateCopy);
+        break;
+      case 'clear':
+        for (const i in stateCopy) {
+          delete stateCopy[i];
+        }
+        addNewState(stateCopy);
```

Comment:
addNewState(stateCopy) is duplicated, so make sense to move it after switch case block